src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -480,6 +481,58 @@ public void ascii() throws IOException {
         );
     }
 
+    @Test
+    public void ifWithThreeParams() throws IOException {
+        assertThat(
+                executeQuery("SELECT if(age > 30, 'True', 'False') AS Ages FROM " + TEST_INDEX_ACCOUNT
+                        + " WHERE age IS NOT NULL GROUP BY Ages", "jdbc"),
+                containsString("\"type\": \"keyword\"")
+        );
+        assertThat(
+                executeQuery("SELECT if(2 > 0, 'hello', 'world') AS iif FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvString("/fields/iif/0", equalTo("hello")))
+        );
+        assertThat(
+                getHits(
+                executeQuery("SELECT iif(2 = 0, 'hello') AS iif FROM " + TEST_INDEX_ACCOUNT)
+                ).getJSONObject(0).query("/fields/iif/0"),
+                equalTo(null)
+        );
+    }
+
+    @Test
+    public void ifWithTwoParams() throws IOException {
+        assertThat(
+                executeQuery("SELECT if(age > 30, 'True') AS Ages FROM " + TEST_INDEX_ACCOUNT
+                        + " WHERE age IS NOT NULL GROUP BY Ages", "jdbc"),
+                containsString("\"type\": \"keyword\"")
+        );
+
+        JSONObject response = executeQuery("SELECT iif(2 = 0, 'hello') AS iif FROM " + TEST_INDEX_ACCOUNT);
+        JSONObject hit = getHits(response).getJSONObject(0);
+        assertThat(
+                ((JSONArray) hit.query("/fields/iif")).get(0),
+                equalTo(null)
+        );
+    }
+
+    @Test
+    public void ifnull() throws IOException {
+        assertThat(
+                executeQuery("SELECT IFNULL(lastname, 'unknown') AS name FROM " + TEST_INDEX_ACCOUNT
+                        + " GROUP BY name", "jdbc"),
+                containsString("\"type\": \"keyword\"")
+        );
+        assertThat(
+                executeQuery("SELECT IFNULL('sample', 'IsNull') AS name FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvString("/fields/name/0", equalTo("sample")))
+        );
+//        assertThat(
+//                executeQuery("SELECT ISNULL(null, 'IsNull') AS name FROM " + TEST_INDEX_ACCOUNT),
+//                hitAny(kvString("/fields/name/0", equalTo("IsNull")))
+//        );
+    }
+
```

Comment:
I commented out this assert because in the query `SELECT IFNULL(null, 'IsNull') ...`, the semantic analyzer seemed to leave the `null` out and recognized the params in ISNULL as one string. But actually it does not matter since typically people will not put `null` as the condition directly into IFNULL condition field. Any suggestions?

Reply:
Could you add the test case of ISNULL.
1) SELECT ISNULL(1+1); expected return 0
2) SELECT ISNULL(1/0); expected return 1
https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_isnull

Reply:
I've added ISNULL(1+1) returns 0, and ISNULL(1+1*1/0) returns 1 into the test. However, when we input a simple enough math expression like `1/0`, the ES throws compile error exception rather than "/ by zero exception" (arithmetic exception) so that it returns "all shards fail". We need to systematically improve our exception mechanism to catch and throw proper exception in the future, so I think this should be part of that as well.